sw/device/lib/testing/test_framework/ottf_main.c

```diff
@@ -90,6 +90,23 @@ static void report_test_status(bool result) {
   test_status_set(result ? kTestStatusPassed : kTestStatusFailed);
 }
 
+// Override the status report function log into on the UART
+void status_report(status_t status) {
```

Comment:
With this implementation, you don't need to maintain a stack in the OTTF do you? you just print as the stack frames pop?

One potential issue, in the future, we may want to be able to print these message encoded in JSON (using the ujson framework) repeatedly a few times, to allow the ATE to pick it up. So it might actually be better to maintain a finite sized stack that you could push the error codes too and print them all at once before the test exits completely, instead of printing them as stack frames pop. Does that makes sense?

Reply:
This was just a quick prototype. The point is that we have a lot of flexiblity in the implementation. For the OTTF, I will implement a small stack and only print it when the test fails. This could then easily be changed to sent it with the UJSON in tests that use this facility.

Reply:
awesome thanks!